### PR TITLE
Internal: Added step for setting up customfonts and colors [ED-15442]

### PR DESCRIPTION
**Pull request by mugurelLupu**
## PR Checklist

- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [x] I have added unittests to verify the code works as intended
- [x] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: core/kits/manager.php [hunk 1]
```diff
@@ -277,7 +277,7 @@ public function get_kit_for_frontend() {
 
 		if ( $is_kit_preview ) {
 			$kit = Plugin::$instance->documents->get_doc_or_auto_save( $active_kit->get_main_id(), get_current_user_id() );
-		} elseif ( 'publish' === $active_kit->get_main_post()->post_status ) {
+		} elseif ( $active_kit->get_main_post() !== null && 'publish' === $active_kit->get_main_post()->post_status ) {
```
Comment (ronenelementor) [inline core/kits/manager.php:280]:
```suggestion
		} elseif ( null !== $active_kit->get_main_post() && 'publish' === $active_kit->get_main_post()->post_status ) {
```

File: modules/checklist/assets/js/app/components/checklist-card-content.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ const ChecklistCardContent = ( { step, setSteps } ) => {
 		learn_more_url: learnMoreUrl,
 		learn_more_text: learnMoreText,
 		image_src: imageSrc,
+		is_locked: isLocked,
```
Comment (ronenelementor) [inline modules/checklist/assets/js/app/components/checklist-card-content.js:16]: Please remove

File: modules/checklist/steps-manager.php [hunk 3]
```diff
@@ -15,7 +16,8 @@
 class Steps_Manager {
 	/** @var Step_Base[] $step_instances */
 	private array $step_instances = [];
-	private static array $step_ids = [ Add_Logo::STEP_ID, Create_Pages::STEP_ID, Setup_Header::STEP_ID, Assign_Homepage::STEP_ID ];
+	private array $steps_config = [];
```
Comment (ronenelementor) [inline modules/checklist/steps-manager.php:19]: Please remove

File: modules/checklist/steps/set-fonts-and-colors.php [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	public function get_description() : string {
+		return esc_html__( 'Global colors and fonts ensure a cohesive look across your site. Start by defining one color and one font.', 'elementor' );
+	}
+
+	public function get_cta_text() : string {
+		return esc_html__( 'Go to Site Identity', 'elementor' );
+	}
+
+	public function get_cta_url() : string {
+		$link = Page::get_site_settings_url_config( true )['url'];
```
Comment (ronenelementor) [inline modules/checklist/steps/set-fonts-and-colors.php:40]: Why bool? The function accepts an id parameter
`get_site_settings_url_config( $active_tab_id = null )`

File: modules/checklist/steps/set-fonts-and-colors.php [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+			'active-document' => Plugin::$instance->kits_manager->get_active_id(),
+		];
+
+		$merged_params = array_merge( $query_params, $additional_params );
+
+		if ( $this->page_exists( $merged_params ) ) {
+			return $this->wordpress_adapter->add_query_arg( $additional_params, $link );
+		}
+
+		return Page::get_create_new_editor_page_url();
```
Comment (ronenelementor) [inline modules/checklist/steps/set-fonts-and-colors.php:59]: This is all handled in `get_site_settings_url_config`

File: modules/checklist/steps/set-fonts-and-colors.php [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+	public function get_learn_more_url() : string {
+		return 'http://go.elementor.com/app-website-checklist-global-article';
+	}
+
+	public function page_exists( $params ) : bool {
+		$query = new \WP_Query( $params );
+
+		return $query->found_posts;
+	}
```
Comment (ronenelementor) [inline modules/checklist/steps/set-fonts-and-colors.php:78]: Unneeded

File: tests/playwright/sanity/plugin-tester-generator.ts [hunk 7]
```diff
@@ -32,7 +32,7 @@ const pluginList: { pluginName: string, installSource: 'api' | 'cli' | 'zip' }[]
 	// { pluginName: 'powerpack-lite-for-elementor', installSource: 'api' },
 	{ pluginName: 'premium-addons-for-elementor', installSource: 'cli' },
 	{ pluginName: 'rife-elementor-extensions', installSource: 'api' },
-	{ pluginName: 'royal-elementor-addons', installSource: 'cli' },
+	//{ pluginName: 'royal-elementor-addons', installSource: 'cli' },
```
Comment (davseve) [inline tests/playwright/sanity/plugin-tester-generator.ts:35]:
```suggestion
	// { pluginName: 'royal-elementor-addons', installSource: 'cli' },
```

File: modules/checklist/assets/js/app/components/checklist-card-content.js (hunk 2 above)
Comment (mugurelLupu) [reply modules/checklist/assets/js/app/components/checklist-card-content.js:16]: Without this i get:  `Uncaught (in promise) ReferenceError: isLocked is not defined
    at _callee$ (checklist-card-content.js:32:94)`

**Review by ronenboxer — DISMISSED** [review]

File: modules/checklist/steps/set-fonts-and-colors.php [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+	public function get_cta_text() : string {
+		return esc_html__( 'Go to Site Identity', 'elementor' );
+	}
+
+	public function get_cta_url() : string {
+		$settings = $this->kit_adapter->get_kit_settings();
+
+		$tab = ! $settings['custom_colors'] ? 'global-typography' : 'global-colors';
+
+		return Page::get_site_settings_url_config($tab)['url'];
```
Comment (louiswol94) [inline modules/checklist/steps/set-fonts-and-colors.php:44]:
```suggestion
		return Page::get_site_settings_url_config( $tab )['url'];
```

**Review by louiswol94 — APPROVED** [review]

**Review by davseve — APPROVED** [review]

**Pull request merged**